### DISCUSSION TITLE
Set Exit code from python shell

### DISCUSSION THREAD

**Discussion by sr1dar** [community/community#28456] [Actions] [answered]
Hi, I have the following piece of code in my workflow.
Is there a way to set exit code (success/failure) without having to create a new env variable or a new step ?
Thanks.

```
    - name: Function Header Check
       if: always()
       run: |
        output=${{needs.build.outputs.output1}}[0]['fun_keyword_check']
        if(output['flag']==1):
          print('pass')
        else:
          print('fail')
          raise Exception('Rule Failed')
       shell: python
```

**Comment by airtower-luna** [comment, marked as answer]
> Is there a way to set exit code (success/failure)

What do you mean by setting the exit code? A run step will be marked as failed if it exits with a non-zero exit code, Python will automatically do that in case of an uncaught exception (like the one you raise in the "fail" case). Alternatively you could use `sys.exit(n)` with `n` as the exit code you want.

If you mean how to avoid creating and checking the output that your step is loading you'd have to look at the job that creates it, assuming that that job failing is sufficient.
